specs/sosreport_integration.adoc

```diff
@@ -0,0 +1,299 @@
+        self.add_cmd_output(["yum repolist | grep tendrl",
+                            "systemctl status tendrl-node-agent.socket"
+                             ])
+
+----
+
+[source, python]
+.tendrl-gluster-integration.py
+----
+from sos.plugins import Plugin
+
+
+class TendrlGlusterIntegration(Plugin):
+    """Tendrl Gluster Integration
+    """
+    plugin_name = "tendrl_gluster_integration"
+    profiles = ('tendrl',)
+
+    def setup(self):
+
+        self.limit = self.get_option("log_size")
+        self.add_copy_spec_limit("/var/log/messages-*", sizelimit=self.limit)
+        self.add_copy_spec("/etc/tendrl/gluster-integration/")
+        self.add_cmd_output("rpm -qa | grep tendrl",
+                            suggest_filename="tendrl_rpm_version")
+        self.add_cmd_output("systemctl status tendrl-node-agent")
+----
+
+[source, python]
+.tendrl-ceph-integration.py
+----
+from sos.plugins import Plugin
+
+
+class TendrlCephIntegration(Plugin):
+    """Tendrl Ceph Integration
+    """
+    plugin_name = "tendrl_ceph_integration"
+    profiles = ('tendrl',)
+
+    def setup(self):
+
+        self.limit = self.get_option("log_size")
+        self.add_copy_spec_limit("/var/log/messages-*", sizelimit=self.limit)
+        self.add_copy_spec("/etc/tendrl/ceph-integration/")
+        self.add_cmd_output("rpm -qa | grep tendrl",
+                            suggest_filename="tendrl_rpm_version")
+        self.add_cmd_output("systemctl status tendrl-node-agent")
+----
+
+[source, python]
+.tendrl-performance-monitoring.py
+----
+from sos.plugins import Plugin
+
+
+class TendrlPerformanceMonitoring(Plugin):
+    """Tendrl Performance Monitoring
+    """
+    plugin_name = "tendrl_performance_monitoring"
+    profiles = ('tendrl',)
+
+    def setup(self):
+
+        self.limit = self.get_option("log_size")
+        self.add_copy_spec_limit("/var/log/messages-*", sizelimit=self.limit)
+        self.add_copy_spec("/etc/tendrl/performance-monitoring/")
+        self.add_cmd_output("rpm -qa | grep tendrl",
+                            suggest_filename="tendrl_rpm_version")
+        self.add_cmd_output(["systemctl status tendrl-node-agent.socket",
+                             "systemctl status carbon-cache",
+                             "ls -la /var/lib/graphite-web/graphite.db"
+                             ])
+
+----
+
+[source, python]
+.tendrl-api.py
+----
+from sos.plugins import Plugin
+
+
+class TendrlApi(Plugin):
+    """Tendrl Node Agent
+    """
+    plugin_name = "tendrl_node_agent"
+    profiles = ('tendrl',)
+
+    def setup(self):
+
+        self.add_copy_spec("/etc/tendrl/etcd.yml")
+        self.add_cmd_output(["ruby -v",
+                             "gem --version"
+                             ])
+        self.add_cmd_output(["systemctl status httpd.service"])
+----
+
+=== Alternatives
+
+* Rather than creating different plugings for different tendrl services, a
+single plugin can also be taken into consideration.
```

Comment:
Please put up a question on the sosreport repo about multi plugin or single plugin approach 
https://github.com/sosreport/sos